src/witan/workspace/onyx.clj

```diff
@@ -188,10 +195,26 @@
 
 (def onyx-catalog-entry-template
   {:onyx/name :witan/name
-   :onyx/fn :witan/fn
+   :onyx/fn (fn [cat config]
+              (if (contains? config :fn-wrapper)
+                (get config :fn-wrapper)
+                (get cat :witan/fn)))
    :onyx/type (constantly :function)
    :onyx/batch-size (fn [_ config]
-                      (get-in config [:batch-settings :onyx/batch-size]))})
+                      (get-in config [:batch-settings :onyx/batch-size]))
+   :onyx/params (fn [cat config]
+                  (->> (conj []
+                             (when (contains? config :fn-wrapper) :witan/fn)
+                             (when (contains? cat :witan/params) :witan/params))
+                       (remove nil?)
+                       (vec)
+                       (not-empty)))
+   ;;;;;;;
+   :witan/params (fn [cat _]
+                   (when (contains? cat :witan/params) (:witan/params cat)))
```

Comment:
can't 

`(when (contains? cat :witan/params) (:witan/params cat))`

be replaced with

`(:witan/params cat)`?